### PR TITLE
Implement Pagination, Axios Configuration with Interceptors, Protected Routes, and Separate Types Folder

### PR DESCRIPTION
**Pull request by batool2azzam**
This pull request includes:

- Pagination: Implemented infinite scrolling for posts.
- Axios Configuration: Added Axios with interceptors for authentication and error handling.
- Protected Routes: Ensured only authenticated users can access certain routes.
- Types Folder: Centralized TypeScript types and interfaces for better code organization.

### CODE REVIEW

File: src/api/api.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    },
+    (error) => {
+        if (error.response.status === 401) {
+            console.error("Unauthorized access - possibly invalid token");
+            localStorage.removeItem("auth_token");
+            window.location.href = "/login";
+        }
+        return Promise.reject(error);
+    }
+);
```
Comment (IbrahimLama) [inline src/api/api.ts]: move these configurations to another file

File: src/components/Posts/Posts.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  };
+
+  useEffect(() => {
+    console.log("Attaching scroll event listener");
+    window.addEventListener("scroll", handleScroll);
+    return () => {
+      console.log("Detaching scroll event listener");
+      window.removeEventListener("scroll", handleScroll);
+    };
+  }, [morePosts, currentPage]);
```
Comment (IbrahimLama) [inline src/components/Posts/Posts.tsx]: why do we need to add the event listener based on this dependency array values ?

File: src/components/Posts/Posts.css [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+}
+
+@keyframes animloader {
+  0% {
+    background-position: 0% 0, 0 0, 70px 5px, 70px 38px, 0px 66px;
+  }
+  100% {
+    background-position: 150% 0, 0 0, 70px 5px, 70px 38px, 0px 66px;
+  }
+}*/
```
Comment (IbrahimLama) [inline src/components/Posts/Posts.css]: remove commented code if not needed

File: src/components/Posts/Posts.tsx (hunk 2 above)
Comment (batool2azzam) [reply src/components/Posts/Posts.tsx]: we doent need it yes

File: src/components/Posts/Posts.css (hunk 3 above)
Comment (batool2azzam) [reply src/components/Posts/Posts.css]: i have delete them

**Pull request merged**